src/analysis/timeline/timelineCredView.js

```diff
@@ -0,0 +1,130 @@
+ * to be contained in the base intervals in TimelineCred
+ */
+export class TimelineCredView {
+  _inputInterval: Interval;
+  _viewIntervals: Interval[];
+  _timelineCred: TimelineCred;
+
+  // We are displaying a range of the intervals in the base TimelineCred
+  // We track the start index and end index of the range, so we can efficiently
+  // slice the cred
+  _startIntervalIndex: number;
+  _endIntervalIndex: number;
+
+  constructor(timelineCred: TimelineCred, inputInterval: ?Interval) {
+    this._timelineCred = timelineCred;
+    if (inputInterval === undefined || inputInterval === null) {
+      this._inputInterval = this._getDefaultInterval(timelineCred);
+    } else {
+      this._inputInterval = inputInterval;
+    }
+
+    this._viewIntervals = weekIntervals(
+      this._inputInterval.startTimeMs,
+      this._inputInterval.endTimeMs
+    );
+
+    this._findIntervalIndices();
+  }
+
+  _findIntervalIndices() {
+    const baseIntervalStrings = new Set(
+      this._timelineCred.intervals().map(stringify)
+    );
+    const viewIntervalStrings = this._viewIntervals.map(stringify);
+    for (const v of viewIntervalStrings) {
+      if (!baseIntervalStrings.has(v)) {
+        throw new Error(`invariant violation: can't find interval ${v}`);
+      }
+    }
+    this._startIntervalIndex = Array.from(baseIntervalStrings).indexOf(
+      viewIntervalStrings[0]
+    );
+    this._endIntervalIndex = Array.from(baseIntervalStrings).indexOf(
+      viewIntervalStrings[viewIntervalStrings.length - 1]
+    );
+
+    if (this._startIntervalIndex < 0 || this._endIntervalIndex < 0) {
+      throw new Error(`invariant violation: input interval 
+        ${stringify(viewIntervalStrings[0]) +
+          "-" +
+          stringify(viewIntervalStrings[viewIntervalStrings.length - 1])} 
+        is not contained in base intervals`);
+    }
+  }
+
+  intervals(): $ReadOnlyArray<Interval> {
+    return this._viewIntervals;
+  }
+
+  /**
+   * Get the CredNode for a given NodeAddress.
+   *
+   * Returns undefined if the node is not in the filtered results.
+   *
+   * Note that it's possible that the node is present in the Graph, but not the
+   * filtered results; if so, it will return undefined.
+   */
+  credNode(a: NodeAddressT): ?CredNode {
+    const fullView = this._timelineCred.getCredByAddress(a);
+    if (fullView === undefined || fullView === null) {
+      return undefined;
+    }
+
+    const cred = fullView.slice(
+      this._startIntervalIndex,
+      this._endIntervalIndex + 1
+    );
+
+    const total = sum(cred);
+    const node = NullUtil.get(this._timelineCred.getNodeByAddress(a));
+    return {cred, total, node};
+  }
+
+  /**
+   * Return all the nodes matching the prefix, along with their cred,
+   * sorted by total cred (descending).
+   */
+  credSortedNodes(prefix: NodeAddressT): $ReadOnlyArray<CredNode> {
+    const match = (a) => NodeAddress.hasPrefix(a, prefix);
+    const addresses: NodeAddressT[] = this._timelineCred
```

Comment:
It's not guaranteed that every node address in the graph will be present in the timelineCred. (timelineCred filters out most of the nodes for space efficiency.) That means that this code will violate its type signature, and return an array containing [undefined].

Instead, we need a method on the base `timelineCred` that returns all the node addresses that it has available.

Consider adding a test that will catch this bug, and then fixing the bug.